artifacts/airgap_patch.py

```diff
@@ -171,93 +129,87 @@ def create_offlineversion_cr():
 
-
-def add_pause_image(origin_image_urls):
-    pause_images = []
-    for image_name in origin_image_urls:
-        if "registry.k8s.io/pause" in image_name:
-            new_version = format(float(image_name.split(":")[1]) - float(0.1), '.1f')
-            pause_images.append("registry.k8s.io/pause:" + new_version)
-    return list(pause_images)
-
-
-image_archs = []
-cni_versions = []
-containerd_versions = []
-kube_versions = []
-calico_versions = []
-cilium_versions = []
-etcd_versions = []
+def get_manifest_data():
+    manifest_keys=['image_arch', 'cni_version', 'containerd_version', 'kube_version', 'calico_version', 'cilium_version', 'etcd_version']
+    manifest_data = {key: [] for key in manifest_keys}
+    manifest_yml_file = os.getenv("MANIFEST_CONF", default="manifest.yml")
+    if (not os.path.exists(manifest_yml_file)) or (Path(manifest_yml_file).read_text().replace("\n", "").strip() == ""):
+        print("manifest yaml file does not exist or empty.")
+        sys.exit(1)
+    with open(manifest_yml_file, "r") as stream:
+        manifest_dict = yaml.safe_load(stream)
+    for key, value in manifest_dict.items():
+        result = []
+        if isinstance(value, str):
+            result.append(value.strip())
+        if isinstance(value, list):
+            for v in value:
+                result.append(str(v).strip())
+        manifest_data[key] = result
+    return manifest_data
+
+def gen_airgap_packages(option, arch, bin_urls, img_urls):
+    if option == "all":
+        create_files(bin_urls, arch=arch)
+        create_images(img_urls, arch=arch)
+        execute_generate_offline_package("copy_import_sh", arch=arch)
+    if option == "create_files":
+        create_files(bin_urls, arch=arch)
+    if option == "create_images":
+        create_images(img_urls, arch=arch)
+
+def batch_gen_airgap_resources(manifest_data, **kwargs):
+    other_required_keywords = ["crictl", "cri-o", "runc", "crun", "runsc", "cri-dockerd", "yq", "coredns", "cluster-proportional-autoscaler"]
+    other_required_list = {key: [] for key in ['file_list', 'image_list']}
+    list_data = {key: [] for key in ['file_list', 'image_list']}
+    is_executed = False
+    for arch in manifest_data.get('image_arch', []):
+        for component, keywords in kwargs.items():
+            for version in manifest_data[component]:
+                os.chdir(SPRAY_REPO_PATH)
+                if os.path.exists("contrib/offline/temp"):
+                    shutil.rmtree("contrib/offline/temp")
+                cmd = ["bash", "contrib/offline/generate_list.sh", "-e", f"{component}='{version}'", "-e", f"image_arch='{arch}'"]
+                result = subprocess.run(cmd, capture_output=True, text=True)
+                if result.returncode != 0:
+                    print(result.stdout)
+                    print(result.stderr)
+                    sys.exit(1)
+                if not os.path.exists("contrib/offline/temp/images.list"):
```

Comment:
`contrib/offline/temp/images.list` appear many times in whole context, it's better to extract as global variable. As `contrib/offline/temp/files.list` should be too.